packages/expo/src/ClerkProvider.tsx

```diff
@@ -25,6 +25,8 @@ export function ClerkProvider(props: ClerkProviderProps): JSX.Element {
   return (
     //@ts-expect-error
     <ClerkReactProvider
+      // Force reset the state when the provided key changes, this ensures that the provider does not retain stale state. See JS-598 for additional context.
+      key={key}
```

Comment:
❓I think that there is `publishableKey` and not `key` prop in `ClerkReactProvider`. Could we double check this?

Reply:
The use of key here is intentional to force react to remount the component and wipe away any stale state from the previous instance. As far as I can tell, by passing the clerk instance below we remove the need to pass the publishableKey as a prop